flink-core/src/main/java/org/apache/flink/types/parser/BooleanParser.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.types.parser;
+
+public class BooleanParser extends FieldParser<Boolean> {
+
+	protected boolean result;
+
+	@Override
+	public int parseField(byte[] bytes, int startPos, int limit, byte[] delim, Boolean reuse) {
+		int i = startPos;
+
+		final int delimLimit = limit - delim.length + 1;
+
+		while (i < limit) {
+			if (i < delimLimit && delimiterNext(bytes, i, delim)) {
+				break;
+			}
+			i++;
+		}
+
+		String str = new String(bytes, startPos, i - startPos);
```

Comment:
The `parseField` might be called a lot of times. 
It would be more efficient if the check would be done directly on the byte array instead of instantiating a String object.